File: pulsar-third/pulsar-chrome-devtools/src/main/java/com/github/kklisura/cdt/protocol/types/network/WebSocketResponse.java
package com.github.kklisura.cdt.protocol.types.network;

/*-
 * #%L
 * cdt-java-client
 * %%
 * Copyright (C) 2018 - 2019 Kenan Klisura
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

import com.github.kklisura.cdt.protocol.support.annotations.Optional;
import java.util.Map;

/** WebSocket response data. */
public class WebSocketResponse {

  private Integer status;

  private String statusText;

  private Map<String, Object> headers;

  @Optional private String headersText;

  @Optional private Map<String, Object> requestHeaders;

  @Optional private String requestHeadersText;

  /** HTTP response status code. */
  public Integer getStatus() {
    return status;
  }

  /** HTTP response status code. */
  public void setStatus(Integer status) {
    this.status = status;
  }

  /** HTTP response status text. */
  public String getStatusText() {
    return statusText;
  }

  /** HTTP response status text. */
  public void setStatusText(String statusText) {
    this.statusText = statusText;
  }

  /** HTTP response headers. */
  public Map<String, Object> getHeaders() {
    return headers;
  }

  /** HTTP response headers. */
  public void setHeaders(Map<String, Object> headers) {
    this.headers = headers;
  }

  /** HTTP response headers text. */
  public String getHeadersText() {
    return headersText;
  }

  /** HTTP response headers text. */
  public void setHeadersText(String headersText) {
    this.headersText = headersText;
  }

  /** HTTP request headers. */
  public Map<String, Object> getRequestHeaders() {
    return requestHeaders;
  }

  /** HTTP request headers. */
  public void setRequestHeaders(Map<String, Object> requestHeaders) {
    this.requestHeaders = requestHeaders;
  }

  /** HTTP request headers text. */
  public String getRequestHeadersText() {
    return requestHeadersText;
  }

  /** HTTP request headers text. */
  public void setRequestHeadersText(String requestHeadersText) {
    this.requestHeadersText = requestHeadersText;
  }
}

File: pulsar-core/pulsar-tools/pulsar-browser/src/main/kotlin/ai/platon/cdt/kt/protocol/types/network/WebSocketResponse.kt
@file:Suppress("unused")
package ai.platon.cdt.kt.protocol.types.network

import ai.platon.cdt.kt.protocol.support.annotations.Optional
import com.fasterxml.jackson.`annotation`.JsonProperty
import kotlin.Any
import kotlin.Int
import kotlin.String
import kotlin.collections.Map

/**
 * WebSocket response data.
 */
data class WebSocketResponse(
  @param:JsonProperty("status")
  val status: Int,
  @param:JsonProperty("statusText")
  val statusText: String,
  @param:JsonProperty("headers")
  val headers: Map<String, Any?>,
  @param:JsonProperty("headersText")
  @param:Optional
  val headersText: String? = null,
  @param:JsonProperty("requestHeaders")
  @param:Optional
  val requestHeaders: Map<String, Any?>? = null,
  @param:JsonProperty("requestHeadersText")
  @param:Optional
  val requestHeadersText: String? = null,
)
